src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -220,6 +220,12 @@ namespace winrt::TerminalApp::implementation
         _root->Initialized({ get_weak(), &TerminalWindow::_pageInitialized });
         _root->WindowSizeChanged({ get_weak(), &TerminalWindow::_WindowSizeChanged });
         _root->RenameWindowRequested({ get_weak(), &TerminalWindow::_RenameWindowRequested });
+        _root->ShowLoadWarningsDialog([weakThis{ get_weak() }](auto&& /*s*/, const Windows::Foundation::Collections::IVectorView<Microsoft::Terminal::Settings::Model::SettingsLoadWarnings>& warnings) {

```

Comment:
in general, this should follow the same pattern as above. Use `{get_weak(), &Foo::Function}` instead of authoring your own weak reference resolver. :)

It will require adding a useless argument to `_ShowLoadWarningsDialog`, but perhaps that's fine